main/src/com/google/refine/model/Recon.java

```diff
@@ -421,4 +423,42 @@ public Recon(
         this.judgmentBatchSize = judgmentBatchSize != null ? judgmentBatchSize : 0;
         this.matchRank = matchRank != null ? matchRank : -1;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + Arrays.deepHashCode(features);
+        result = prime * result + Objects.hash(candidates, error, id, identifierSpace, judgment, judgmentAction,
+                judgmentBatchSize, judgmentHistoryEntry, match, matchRank, schemaSpace, service);
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Recon other = (Recon) obj;
+        return Objects.equals(candidates, other.candidates) && Objects.equals(error, other.error)
```

Comment:
Is this equivalent to `Objects.deepEquals()`? If yes, please simplify. If no, please add a comment about which fields are (intentionally) left out. I think all fields are included, but I easily could have missed something.

Reply:
Do you mean [this Objects.deepEquals()](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#deepEquals-java.lang.Object-java.lang.Object-)? It's not clear to me how using this would simplify this method. I don't think I can replace the entire method code by just a call to `deepEquals` since according to its documentation, it would just call back this very `equals` method, resulting in an infinite recursion. What sort of simplification did you have in mind?

Reply:
Yes, that was dumb of me! That method doesn't do what I thought it did.